partner_email_check/models/res_partner.py

```diff
@@ -9,30 +9,84 @@
 
 try:
-    from validate_email import validate_email
+    from email_validator import validate_email, EmailNotValidError
 except ImportError:
-    _logger.debug('Cannot import "validate_email".')
+    _logger.debug('Cannot import "email_validator".')
 
-    def validate_email(email):
-        _logger.warning(
-            'Can not validate email, '
-            'python dependency required "validate_email"')
-        return True
+    validate_email = None
 
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    @api.constrains('email')
-    def constrains_email(self):
-        for rec in self.filtered("email"):
-            self.email_check(rec.email)
-
     @api.model
     def email_check(self, emails):
-        for email in emails.split(','):
-            if not validate_email(email):
-                raise ValidationError(
-                    _("%s is an invalid email") % email.strip()
-                )
-        return True
+        return ','.join(self._normalize_email(email.strip())
+                        for email in emails.split(','))
+
+    def _normalize_email(self, email):
+        if validate_email is None:
+            _logger.warning(
+                'Can not validate email, '
+                'python dependency required "email_validator"')
+            return email
+
+        try:
+            result = validate_email(
+                email,
+                check_deliverability=self._should_check_deliverability(),
+            )
+        except EmailNotValidError as e:
+            raise ValidationError(
+                _("%s is an invalid email: %s") % (email.strip(), e.message)
+            )
+        return result['local'].lower() + '@' + result['domain_i18n']
+
+    def _should_filter_duplicates(self):
+        conf = self.env['ir.config_parameter'].get_param(
+            'partner_email_check_filter_duplicates', 'False'
+        )
+        return conf == 'True'
+
+    def _should_check_deliverability(self):
+        conf = self.env['ir.config_parameter'].get_param(
+            'partner_email_check_check_deliverability', 'False'
+        )
+        return conf == 'True'
+
+    def _validate_no_duplicates(self, email):
+        is_existing_record = bool(self)
+        if is_existing_record:
+            self.ensure_one()
+
+        if is_existing_record and email == self.email:
+            # No change, no need to check
+            return
+
+        if ',' in email:
+            raise ValidationError(
+                _("Field contains multiple email addresses. This is not "
+                  "supported when duplicate email addresses are not allowed.")
+            )
+
+        domain = [('email', '=', email)]
+        if is_existing_record:
+            domain.append(('id', '!=', self.id))
+        if self.search(domain, limit=1):
+            raise ValidationError(_("Email '%s' is already in use.") % email)
+
+    @api.model
+    def create(self, vals):
+        if vals.get('email', False):
```

Comment:
no need to pass a default here, as otherwise you get `None` and that's falsy too, which is all you need